lib/src/Uncertainty/Algorithm/Sensitivity/openturns/RankSobolSensitivityAlgorithm.hxx

```diff
@@ -0,0 +1,149 @@
+                 const Sample & outputDesign);
+                                 
+  /** First order indices accessor */
+  Point getFirstOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+  
+  /** String converter */
+  String __repr__() const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;  
+  
+  /** Second order indices accessor */
+  SymmetricMatrix getSecondOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+  
+  /** Total order indices accessor */
+  Point getTotalOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+
+  /** Interval for the total order indices accessor */
+  Interval getTotalOrderIndicesInterval() const override;
+  
+  /** Aggregated total order indices accessor for multivariate samples */
+  Point getAggregatedTotalOrderIndices() const override;
+  
+  /** Whether to use bootstrap or asymptotic distribution */
+  void setUseAsymptoticDistribution(Bool ) override {};
+  Bool getUseAsymptoticDistribution() const override;
+
+  /** Estimator distribution accessor */
+  Distribution getTotalOrderIndicesDistribution() const override;
+   
+protected:
+
+  Sample computeIndices() const;
+  
+  /** void method that computes bootstrap confidence interval */
+  void computeBootstrapDistribution() const override;
+
+private:
+
+  static Graph DrawSobolFirstIndices(const Description & inputDescription,
+                                     const Point & firstOrderIndices,
+                                     const Interval & firstOrderConfidenceInterval = Interval());
+
+  /** Internal method that compute Vi/VTi using a sample */
+  Sample computeIndices(const Sample & sample, Sample & VTi) const override;
+  
+  Sample computeIndicesFromSample(const Sample &, const Sample &) const;
+  
+  /** Function that computes aggregated indices using Vi/VTi + variance  */
+  Point computeAggregatedIndices(const Sample & Vi,
+                                 const Point & variance) const;
+                                 
+  /** Constructor with distribution / model parameters */
+  RankSobolSensitivityAlgorithm(const Distribution & ,
+                                const UnsignedInteger ,
+                                const Function & ,
+                                const Bool ){};
+
+  /** Constructor with experiment / model parameters */
+  RankSobolSensitivityAlgorithm(const WeightedExperiment & ,
+                                const Function & ,
+                                const Bool ){};
+                                
+  /** Constructor with parameters */
+  RankSobolSensitivityAlgorithm(const Sample & ,
+                                const Sample & , 
+                                const UnsignedInteger ){};
+                                
```

Comment:
these ctors are declared but not defined ? drop them ?

Reply:
but they are defined as public in the mother class. We move them to private in order not to be available for this class

Reply:
they wont be available anyway (ctors are not inherited), so I think they can be dropped